File: code-examples/cpp/usage-examples/updateOne.cpp
#include <bsoncxx/builder/stream/document.hpp>
#include <bsoncxx/json.hpp>
#include <mongocxx/client.hpp>
#include <mongocxx/stdx.hpp>
#include <mongocxx/uri.hpp>
#include <iostream>
#include <random>

using bsoncxx::builder::stream::document;

int main() {
    std::string uri = "<connection string uri>";
    mongocxx::client client{mongocxx::uri{uri}};

    try {
        auto database = client["sample_mflix"];
        auto movies = database["movies"];

        /* Generate some random double to simulate random number plot */
        std::random_device dev;
        std::mt19937 rng(dev());
        std::uniform_real_distribution<> dist(0.0, 1.0);
        auto randomPlot = "A harvest of random numbers, such as: " + std::to_string(dist(rng));
        
        auto filter_doc = document{} << "title" << "Random Harvest" << bsoncxx::builder::stream::finalize;
        auto update_doc = document{} << "$set" << bsoncxx::builder::stream::open_document 
                                      << "plot" << randomPlot << bsoncxx::builder::stream::close_document 
                                      << bsoncxx::builder::stream::finalize;

        // options for update operation
        mongocxx::options::update options;
        options.upsert(true);

        // Perform update operation
        auto result = movies.update_one(filter_doc.view(), update_doc.view(), options);

        std::cout << result->matched_count() << " document(s) matched the filter, updated " << result->modified_count() << " document(s)\n";
    } catch (const std::exception &e) {
        std::cerr << e.what() << "\n";
    }

    // Close the connection
    client.~client();

    return 0;
}

File: code-examples/pymongo/usage-examples/updateOne.py
# Update a document

from pymongo import MongoClient
import random 

# Replace the uri string with your MongoDB deployment's connection string
uri = "<connection string uri>"

client = MongoClient(uri)

def run():
    try:
       
        database = client["sample_mflix"]
        movies = database["movies"]

        # Update a document that has the title "Random Harvest" to have a plot field with the specified value
        result = movies.update_one(
          { "title": "Random Harvest" },
          {
            "$set": {
              "plot": "A harvest of random numbers, such as: {}".format(random.random()),
            },
          },
          # Set the upsert option to insert a document if no documents match the filter
          upsert=True
        )

        # Print the number of matching and modified documents
        print(
          "{} document(s) matched the filter, updated {} document(s)".format(result.matched_count, result.modified_count)
        )
    except Exception as e:
        print(e)
    finally:
        # Close the connection after the operation completes
        client.close()

# Run the program and print any thrown errors
run()